src/cli/cli_coap.cpp

```diff
@@ -0,0 +1,356 @@
+                          HostSwap16(aMessageInfo->mSockAddr.mFields.m16[3]),
+                          HostSwap16(aMessageInfo->mSockAddr.mFields.m16[4]),
+                          HostSwap16(aMessageInfo->mSockAddr.mFields.m16[5]),
+                          HostSwap16(aMessageInfo->mSockAddr.mFields.m16[6]),
+                          HostSwap16(aMessageInfo->mSockAddr.mFields.m16[7]));
+
+    switch (otCoapHeaderGetCode(aHeader))
+    {
+    case kCoapRequestGet:
+        sServer->OutputFormat("GET");
+        break;
+
+    case kCoapRequestDelete:
+        sServer->OutputFormat("DELETE");
+        break;
+
+    case kCoapRequestPut:
+        sServer->OutputFormat("PUT");
+        break;
+
+    case kCoapRequestPost:
+        sServer->OutputFormat("POST");
+        break;
+
+    default:
+        sServer->OutputFormat("Undefined\r\n");
+        return;
+    }
+
+    PrintPayload(aMessage);
+
+    if ((otCoapHeaderGetType(aHeader) == kCoapTypeConfirmable) || otCoapHeaderGetCode(aHeader) == kCoapRequestGet)
+    {
+        otCoapHeader responseHeader;
+        otCoapCode responseCode = kCoapResponseValid;
+        char responseContent = '0';
+
+        if (otCoapHeaderGetCode(aHeader) == kCoapRequestGet)
+        {
+            responseCode = kCoapResponseContent;
+        }
+
+        otCoapHeaderInit(&responseHeader, kCoapTypeAcknowledgment, responseCode);
+        otCoapHeaderSetMessageId(&responseHeader, otCoapHeaderGetMessageId(aHeader));
+        otCoapHeaderSetToken(&responseHeader, otCoapHeaderGetToken(aHeader), otCoapHeaderGetTokenLength(aHeader));
+
+        if (otCoapHeaderGetCode(aHeader) == kCoapRequestGet)
+        {
+            otCoapHeaderSetPayloadMarker(&responseHeader);
+        }
+
+        responseMessage = otCoapNewMessage(sInstance, &responseHeader);
+        VerifyOrExit(responseMessage != NULL, error = kThreadError_Failed);
+
+        if (otCoapHeaderGetCode(aHeader) == kCoapRequestGet)
+        {
+            SuccessOrExit(error = otMessageAppend(responseMessage, &responseContent, sizeof(responseContent)));
+        }
+
+        SuccessOrExit(error = otCoapSendResponse(sInstance, responseMessage, aMessageInfo));
+    }
+
+exit:
+
+    if (error != kThreadError_None && responseMessage != NULL)
+    {
+        sServer->OutputFormat("Cannot send CoAP response message: Error %d\r\n", error);
+        otMessageFree(responseMessage);
+    }
+    else
+    {
+        if ((otCoapHeaderGetType(aHeader) == kCoapTypeConfirmable) || otCoapHeaderGetCode(aHeader) == kCoapRequestGet)
+        {
+            sServer->OutputFormat("CoAP response sent successfully!\r\n");
+        }
+    }
+
+    (void)aHeader;
+    (void)aMessage;
+    (void)aMessageInfo;
+}
+
+ThreadError Coap::ProcessClient(int argc, char *argv[])
+{
+    ThreadError error = kThreadError_None;
+    otMessage *message;
+    otMessageInfo messageInfo;
+    otCoapHeader header;
+
+    // Default parameters
+    char coapUri[] = "test";
```

Comment:
Looking at this again, you'll want to do a couple things:
1. Specify the maximum input length to support.
2. Use the safe version of `strcpy()`, namely `strlcpy()`.